Improve datamapper documentation (#713)
wso2/docs-mi: en/docs/reference/mediators/data-mapper-mediator.md

Reviewer: The images on this page are currently named with numbers instead of meaningful, lowercase names separated by hyphens. While this follows the initial approach used when the page was created, it is not the recommended naming convention for images. Shall we update the image names and adjust the respective references in the markdown file accordingly?

Author: Ack. Renamed the image files and removed the obsolete images from the directory.

Reviewer: The alt texts for most of the images are currently empty (`[]`). To improve accessibility and provide a better user experience, let's add meaningful alt tags inside the brackets.

Author: Ack. Added alt texts for the images.

# Data Mapper Mediator

The Data Mapper mediator is a data mapping solution that can be integrated
into a mediation sequence. It converts and transforms one data format to

> another or changes the structure of the data in a message. The WSO2 MI VSCode extension

Reviewer: ```suggestion
another or changes the structure of the data in a message. The WSO2 Micro Integrator for Visual Studio Code (MI for VS Code) extension
```

provides a graphical mapping configuration and
generates the files required for executing this graphical mapping
configuration through the WSO2 Data Mapper engine.

> The Data Mapper component utilizes TypeScript to define the mapping when you use the WSO2 MI VSCode Extension. When you build the Integration project using the WSO2 MI VSCode Extension, the Data Mapper component generates the required configuration files needed by the Data Mapper Engine to execute the mapping. These files are stored in the Governance Registry.

Reviewer: ```suggestion
The Data Mapper component utilizes TypeScript to define the mapping when you use the WSO2 MI for VS Code extension. When you build the Integration project using the WSO2 MI for VS Code extension, the Data Mapper component generates the required configuration files needed by the Data Mapper Engine to execute the mapping. These files are stored in the Governance Registry.
```

<tbody>
<tr class="odd">
<td><strong>Mapping Configuration</strong></td>

> <td>The file, which contains the script file that is used to execute the mapping. This file is generated when you build the Integration project using WSO2 MI VSCode extension. It is stored under <b>Governance Registry</b>.</td>

Reviewer: ```suggestion
<td>The file, which contains the script file that is used to execute the mapping. This file is generated when you build the Integration project using the WSO2 MI for VS Code extension. It is stored under <b>Governance Registry</b>.</td>
```

</tr>
<tr class="even">
<td><strong>Input Schema</strong></td>

> <td>JSON schema, which represents the input message format. This file is generated when you build the Integration project using WSO2 MI VSCode extension. It is stored under <b>Governance Registry</b>.</td>

Reviewer: ```suggestion
<td>JSON schema, which represents the input message format. This file is generated when you build the Integration project using the WSO2 MI for VS Code extension. It is stored under <b>Governance Registry</b>.</td>
```

</td>
</tr>
<tr class="odd">
<td><strong>Output Schema</strong></td>

> <td>JSON schema, which represents the output message format. This file is generated when you build the Integration project using WSO2 MI VSCode extension. It is stored under <b>Governance Registry</b>.</td>

Reviewer: ```suggestion
<td>JSON schema, which represents the output message format. This file is generated when you build the Integration project using the WSO2 MI for VS Code extension. It is stored under <b>Governance Registry</b>.</td>
```

</tr>
<tr class="even">
<td><strong>Input Type</strong></td>
<td>Expected input message type (XML/JSON/CSV).</br>
<div class="admonition note">
<p class="admonition-title">Note</p>

> <p>By default, the Input type is selected from the Input Schema if you have not provided. If you want to specify Input type, then you can specify it in WSO2 MI VSCode Extension.</p>

Reviewer: ```suggestion
<p>By default, the Input Type is selected from the Input Schema if you have not provided one. If you want to specify the Input Type, then you can specify it in the WSO2 MI for VS Code extension.</p>
```

</div>

</td>
</tr>
<tr class="odd">
<td><strong>Output Type</strong></td>
<td>Target output message type (XML/JSON/CSV).</br>
<div class="admonition note">
<p class="admonition-title">Note</p>

> <p>By default, the Output type is selected from the Output Schema if you have not provided. If you want to specify Output type, then you can specify it in WSO2 MI VSCode Extension.</p>

Reviewer: ```suggestion
<p>By default, the output type is selected from the output schema if you have not provided one. If you want to specify the output type, you can specify it in the WSO2 MI for VS Code extension.</p>
```

</div></td>
</tr>
</tbody>
</table>

## Components of Data Mapper

### Mapping configuration file

> The Data Mapper component utilizes TypeScript to define the mapping when you use the WSO2 MI VSCode Extension. When you build the integration project using the WSO2 MI VSCode Extension, the data mapping configuration file is compiled into JavaScript, which is used by the Data Mapper engine to map the elements in the MI runtime.

Reviewer: ```suggestion
The Data Mapper component utilizes TypeScript to define the mapping when you use the WSO2 MI for VS Code extension. When you build the integration project using the WSO2 MI for VS Code extension, the data mapping configuration file is compiled into JavaScript, which is used by the Data Mapper engine to map the elements in the MI runtime.
```

> ![]({{base_path}}/assets/img/integrate/mediators/119131284/119135199.png) 

Reviewer: Shall we add a meaningful alt tag here?

### Input and output files

You can load the following input/output message formats:

-   **XML:** to load a sample XML file
-   **JSON:** to load a sample JSON file
-   **CSV:** to load a sample CSV file with column names as the first
    record
-   **JSONSCHEMA:** to load a JSON schema

![]({{base_path}}/assets/img/integrate/mediators/119131284/119134796.png) 

> As per the provided input/output files, the Typescript interfaces are generated inside the Mapping configuration file. JSON schema files are generated when you build a project using the WSO2 MI VSCode Extension. These files are stored in the CAR file and deployed inside the governance registry. These files are used to validate the input and output messages at runtime.

Reviewer: ```suggestion
Based on the provided input/output files, the Typescript interfaces are generated inside the Mapping configuration file. JSON schema files are generated when you build a project using the WSO2 MI for VS Code extension. These files are stored in the CAR file and deployed inside the governance registry. These files are used to validate the input and output messages at runtime.
```

!!! info "Reimport input/output files"
    If you need to reimport input/output, you can right click on top of the input/output widget. 

!!! info "Filter elements"

>     If you need to filter elements in input/output, you can use the filter bar on right side upper corner of the Data mapper view.

Reviewer: ```suggestion
    If you need to filter elements in input/output, you can use the filter bar on the upper right corner of the Data Mapper view.
```

SOAP body elements to create a certain SOAP payload, by creating a
Salesforce convertLead SOAP payload using a JSON payload. The Convert
Lead SOAP payload needs mapping SOAP header information.  

> E.g. `<urn:sessionId>QwWsHJyTPW.1pd0_jXlNKOSU</urn:sessionId>`

Reviewer: ```suggestion
For example: `<urn:sessionId>QwWsHJyTPW.1pd0_jXlNKOSU</urn:sessionId>`
```

> boolean etc. values into a JSON payload with required primitive types,

Reviewer: ```suggestion
boolean and other values into a JSON payload with required primitive types,
```

This example demonstrates how you can map an XML payload to CSV format.

!!! Info

>     If you specify special characters (e.g., `&` ,

Reviewer: ```suggestion
    If you specify special characters (For example `&`,
```

>     `&amp;)` within the `<text>`

Reviewer: Is the bracket placed correctly here?

Author: Fixed it.

>     tag when converting from CSV to CSV , they will be displayed as follows

Reviewer: ```suggestion
    tag when converting from CSV to CSV, they will be displayed as follows
```
